app/controllers/bookings_controller.rb

```diff
@@ -0,0 +1,44 @@
+class BookingsController < ApplicationController
+  before_action :set_event
+  before_action :set_booking, only: %i[update cancelation]
+
+  def create
+    @booking = @event.bookings.new(booking_params)
+
+    if @booking.save
+      redirect_to root_path
+    else
+      render "pages/home", status: :unprocessable_entity
+    end
+  end
+
+  def update
+    if @booking.update(booking_params)
+      redirect_to root_path
+    else
+      render "pages/home", status: :unprocessable_entity
+    end
+  end
+
+  def cancelation
+    if @booking.canceled?
+      @booking.update(status: "canceled")
+    else
+      render "pages/home", status: :unprocessable_entity
+    end
+  end
+
+  private
+
+  def set_booking
+    @booking = Booking.find(params[:id])
+  end
+
+  def set_event
+    @event = Event.find(params[:event_id])
+  end
+
+  def booking_params
+    params.require(:booking).permit(:canceled)
+  end
```

Comment:
We don't want to permit cancel in the form.
Do you think it'd be alright to let a client create a new booking by saying the canceled is true ? 😄 
Beside there won't even be a form so we don't need any params, you can delete this method

Reply:
deleted